### PR TITLE
Add type definitions for object-map.

### PR DESCRIPTION
**Pull request by wolfgang42**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).
- [x] The package does not provide its own types, and you can not add them.
- [x] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] `tslint.json` should be present, and `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, and `strictNullChecks` set to `true`.

### CODE REVIEW

**Comment by wolfgang42** [comment]
Travis apparently runs some extra linting that I wasn't aware of. I don't see this in the Readme but maybe I missed something. Replacement commit pushed with fix.

File: types/object-map/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,12 @@
+// Type definitions for object-map 1.0
+// Project: https://github.com/xixixao/object-map
+// Definitions by: Wolfgang Faust <https://github.com/wolfgang42>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+declare function objectMap<I, O>(
```
Comment (plantain-00) [inline types/object-map/index.d.ts]: I suggest not to use `I`, `O`, it looks like `1`, `0`

**Review by plantain-00 — CHANGES_REQUESTED** [review]

File: types/object-map/index.d.ts (hunk 1 above)
Comment (wolfgang42) [reply types/object-map/index.d.ts]: What do you suggest using instead? I picked them for `I`nput and `O`utput but I'm open to other suggestions.

Comment (plantain-00) [reply types/object-map/index.d.ts]: I believe `TInput` and `TOutput` is OK.

**Review by plantain-00 — APPROVED** [review]
The CI error seems not related to the code change.

**Comment by wolfgang42** [comment]
Is there anything more I need to do before this can be merged? I don't seem to have permission to retry the Travis build.

**Comment by plantain-00** [comment]
@andy-ms the CI errored(not failed), can you check on it?

File: types/object-map/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,12 @@
+// Type definitions for object-map 1.0
+// Project: https://github.com/xixixao/object-map
+// Definitions by: Wolfgang Faust <https://github.com/wolfgang42>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+declare function objectMap<TInput, TOutput>(
+	target: {[k: string]: TInput},
+	callback: (currentValue: TInput, key: string, object: {[k: string]: TInput}) => TOutput,
+	thisArg?: any
```
Comment (andy-ms) [inline types/object-map/index.d.ts]: If you really want to support a `thisArg` you might want an overload using `callback: (this: TThis, ...) => TOutput`.

Comment (wolfgang42) [reply types/object-map/index.d.ts]: Cool, I didn't realize that was possible, I didn't see anything about declaring `this` when I looked in the TS documentation. I'll push a new version in a few minutes. Thanks!

**Pull request merged**